x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/SqlProtocolTestCase.java

```diff
@@ -195,6 +195,18 @@ private void assertQuery(String sql, String columnName, String columnType, Objec
             requestContent = new StringBuilder(requestContent)
                     .insert(requestContent.length() - 1, ",\"columnar\":" + columnar).toString();
         }
+
+        // randomize binary response enforcement for drivers (ODBC/JDBC) and CLI
+        boolean binaryResponse = randomBoolean();
+        Mode m = Mode.fromString(mode);
+        if (randomBoolean()) {
+            // set it explicitly or leave the default (null) as is
```

Comment:
I'm confused, where is the `null` in this randomisation?

Reply:
`null` means the parameter is not set at all (no explicit `false` or `true` value).

Reply:
Thanks, somehow I missed the whole logic there.